pkg/images/images.go

```diff
@@ -0,0 +1,203 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package images
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/DataDog/datadog-operator/api/datadoghq/v2alpha1"
+)
+
+const (
+	// AgentLatestVersion corresponds to the latest stable agent release
+	AgentLatestVersion = "7.64.3"
+	// ClusterAgentLatestVersion corresponds to the latest stable cluster-agent release
+	ClusterAgentLatestVersion = "7.64.3"
+	// FIPSProxyLatestVersion corresponds to the latest stable fips-proxy release
+	FIPSProxyLatestVersion = "1.1.10"
+	// GCRContainerRegistry corresponds to the datadoghq GCR registry
+	GCRContainerRegistry = "gcr.io/datadoghq"
+	// DockerHubContainerRegistry corresponds to the datadoghq docker.io registry
+	DockerHubContainerRegistry = "docker.io/datadog"
+	// PublicECSContainerRegistry corresponds to the datadoghq PublicECSContainerRegistry registry
+	PublicECSContainerRegistry = "public.ecr.aws/datadog"
+	// DefaultImageRegistry corresponds to the datadoghq containers registry
+	DefaultImageRegistry = "gcr.io/datadoghq"
+	// Default Image Registries
+	DefaultAzureImageRegistry  string = "datadoghq.azurecr.io"
+	DefaultEuropeImageRegistry string = "eu.gcr.io/datadoghq"
+	DefaultAsiaImageRegistry   string = "asia.gcr.io/datadoghq"
+	DefaultGovImageRegistry    string = "public.ecr.aws/datadog"
+	// JMXTagSuffix suffix tag for agent JMX images
+	JMXTagSuffix = "-jmx"
+	// FIPSTagSuffix suffix tag for agent FIPS images
+	FIPSTagSuffix = "-fips"
+	// Default Image names
+	DefaultAgentImageName        string = "agent"
+	DefaultClusterAgentImageName string = "cluster-agent"
+	OTelAgentBetaTag                    = "7.63.0-ot-beta-jmx"
+)
+
+// imageHasTag identifies whether an image string contains a tag suffix
+// Ref: https://github.com/distribution/distribution/blob/v2.7.1/reference/reference.go
+var imageHasTag = regexp.MustCompile(`.+:[\w][\w.-]{0,127}$`)
+
+// imageNameContainsTag return true if the image name contains a tag
+func imageNameContainsTag(name string) bool {
+	// The image name corresponds to a full image string
+	return imageHasTag.MatchString(name)
+}
+
+// Image represents a container image information
+type Image struct {
+	registry string
+	name     string
+	tag      string
+	isJMX    bool
+	isFIPS   bool
+}
+
+// NewImage return a new Image instance
+// Assumes that tag suffixes have already been trimmed and booleans updated accordingly
+func NewImage(registry, name, tag string, isJMX bool, isFIPS bool) *Image {
```

Comment:
```suggestion
func newImage(registry, name, tag string, isJMX bool, isFIPS bool) *Image {
```
no need to export this.